### PR TITLE
CASMNET-1549: Add sshd_config cloud-init step and template

### PR DESCRIPTION
**Pull request by manderson-hpe**
#### Summary and Scope

- Fixes [CASMNET-1549](https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1549)

##### Issue Type

- Bugfix Pull Request

Allows access to sshd on any network that is not CHN and CAN.

#### Prerequisites

- [ ] I have included documentation in my PR (or it is not required)
- [x] I tested this on metal (e.g. an internal system, with hardware) (x) (if yes, please include results or a description of the test)
- [ ] I tested this on vshasta (if yes, please include results or a description of the test)

### CODE REVIEW

File: boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## template:jinja
+# Your system has been configured with 'manage-sshd_config' set to true.
```
Comment (rustydb) [inline boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl:2]: Where is this set? The function you added looks like it always runs.

File: boxes/ncn-common/files/scripts/metal/net-init.sh [hunk 2]
```diff
@@ -112,6 +112,20 @@ function ifconf() {
 }
 ifconf
 
+function sshd_config() {
```
Comment (rustydb) [inline boxes/ncn-common/files/scripts/metal/net-init.sh:115]: If you want this in net init, you should move this _after_ the IP test in the next block.

File: boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# As a result, cloud-init has written this file with configuration data
+# that it has been provided. Cloud-init, by default, will write this file
+# a single time (PER_ONCE).
+# Default SUSE/sshd_config with all empty-lines and comments (asides this ones) removed.
+AcceptEnv 			LANG LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES
+AcceptEnv 			LC_IDENTIFICATION LC_ALL
+AcceptEnv 			LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT
+AuthorizedKeysFile  .ssh/authorized_keys
+PermitRootLogin 	yes
```
Comment (rustydb) [inline boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl:11]: nit: all the things on the right side are wildy spaced different, can you align the values?

**Review by rustydb — CHANGES_REQUESTED** [review]

File: boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl (hunk 1 above)
Comment (manderson-hpe) [reply boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl:2]: It does always run. I'm either going to remove that or figure out where those vars are set and update it.

File: boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl (hunk 3 above)
Comment (manderson-hpe) [reply boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl:11]: This appears to be a github things. The editor it looks fine, but I think it is just the usage of tabs. 

image

Comment (manderson-hpe) [reply boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl]: I removed the spacing and made it look like the default config file. Which is more consistent and easier to work with in code. It'll also be better this way for ansible management.

File: boxes/ncn-common/files/scripts/metal/net-init.sh (hunk 2 above)
Comment (manderson-hpe) [reply boxes/ncn-common/files/scripts/metal/net-init.sh]: I have moved it to after the IP test but before the line that says it's all done.

File: boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl (hunk 1 above)
Comment (manderson-hpe) [reply boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl]: The only reference I could find for where I took this from is: https://github.com/Cray-HPE/metal-cloud-init/blob/41f93a6000406ca974a5b9f5dc5e29f97bb59edb/doc/examples/cloud-config-resolv-conf.txt so I have removed it.

Comment (rustydb) [reply boxes/ncn-common/files/resources/common/cloud/templates/sshd_config.tmpl]: Cool. The resolv and hosts files are cloud init toggles, I hadnt heard of the sshd one so I asked.

**Review by rustydb — APPROVED** [review]

**Comment by jacobsalmela** [comment]
Did you test a booted image?  I have noticed the variables in the templates sometimes differ slightly.

**Comment by manderson-hpe** [comment]
> Did you test a booted image? I have noticed the variables in the templates sometimes differ slightly.

No, I haven't tested a booted image. That's my goal when we can get a build working. As you pointed out, I presume it will fail, but I guess I won't know until I run it on a real system to see what the value are.

**Pull request merged**